appveyor.yml

```diff
@@ -0,0 +1,124 @@
+platform: x64
+environment:
+ matrix:
+  - DC: dmd
+    DVersion: nightly
+    arch: x64
+  - DC: dmd
+    DVersion: nightly
+    arch: x86
+  - DC: dmd
+    DVersion: beta
+    arch: x64
+  - DC: dmd
+    DVersion: beta
+    arch: x86
+  - DC: dmd
+    DVersion: stable
+    arch: x64
+  - DC: dmd
+    DVersion: 2.072.2
+    arch: x86
+  - DC: dmd
+    DVersion: 2.072.2
+    arch: x64
+  - DC: dmd
+    DVersion: stable
+    arch: x86
+  - DC: ldc
+    DVersion: beta
+    arch: x64
+  - DC: ldc
+    DVersion: stable
+    arch: x64
+  - DC: ldc
+    DVersion: 1.2.0
+    arch: x64
+
+skip_tags: false
+branches:
+  only:
+    - master
+
+install:
+  - ps: function ResolveLatestDMD
+        {
+            $version = $env:DVersion;
+            if($version -eq "stable") {
+                $latest = (Invoke-WebRequest "http://downloads.dlang.org/releases/LATEST").toString();
+                $url = "http://downloads.dlang.org/releases/2.x/$($latest)/dmd.$($latest).windows.7z";
+            }elseif($version -eq "beta") {
+                $latest = (Invoke-WebRequest "http://downloads.dlang.org/pre-releases/LATEST").toString();
+                $latestVersion = $latest.split("-")[0].split("~")[0];
+                $url = "http://downloads.dlang.org/pre-releases/2.x/$($latestVersion)/dmd.$($latest).windows.7z";
+            }elseif($version -eq "nightly") {
+                $url = "http://nightlies.dlang.org/dmd-master-2017-05-20/dmd.master.windows.7z"
```

Comment:
http://nightlies.dlang.org/dmd-nightly/LATEST.
In fact you want to redirect any named branch to http://nightlies.dlang.org/dmd-<name>/LATEST, then download http://nightlies.dlang.org/dmd-<result>/dmd.<result-branch>.windows.7z.
A 2-level lookup as in install.sh should work best.
https://github.com/dlang/installer/blob/efbba50ad7102bf1eb5f25c5886b5cf128cbcaea/script/install.sh#L397
First resolving the specific version, the downloading that.
That also allows to directly download a specific build.